pkg/gardenlet/controller/shoot/shoot_care_control.go

```diff
@@ -177,6 +178,11 @@ func (c *defaultCareControl) Care(shootObj *gardencorev1beta1.Shoot, key string)
 		return fmt.Errorf("failed to get garden client: %w", err)
 	}
 
+	gardenerVersion, err := semver.NewVersion(shoot.Status.Gardener.Version)
```

Comment:
Can we add a `GardenerSemver` member or something similar to the `shoot.Shoot` type?
I guess, this has potential for reuse :)